library/src/main/scala/scala/scalajs/runtime/RuntimeLong.scala

```diff
@@ -304,7 +304,29 @@ final class RuntimeLong(val lo: Int, val hi: Int)
   def *(b: RuntimeLong): RuntimeLong = {
     val alo = a.lo
     val blo = b.lo
-    new RuntimeLong(alo * blo, RuntimeLong.timesHi(alo, a.hi, blo, b.hi))
+
+    val a0 = alo & 0xffff
+    val a1 = alo >>> 16
+    val b0 = blo & 0xffff
+    val b1 = blo >>> 16
+
+    val a0b0 = a0 * b0
+    val a1b0 = a1 * b0 // collapses to 0 when a is constant and 0 <= a <= 0xffff
```

Comment:
Consider formulating this comment in a way that makes it explicit that it applies equally to `a0b1` and `b`.